independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/PreResponseFilterHandler.java

```diff
@@ -0,0 +1,15 @@
+package org.jboss.resteasy.reactive.client.handlers;
+
+import org.jboss.resteasy.reactive.client.impl.RestClientRequestContext;
+import org.jboss.resteasy.reactive.client.spi.ClientRestHandler;
+
+/**
+ * This Handler is invoked before ClientResponseFilters handler. It changes the abort handler chain
+ * to a one without the ClientResponseFilterRestHandler's so that the filters are not retriggered in case of failure
```

Comment:
I just looked at the server case and in that case (in accordance to the spec) the following happens:

If there is an exception mapper, then the response filters **are** executed.
If there is no exception mapper, then the response filter are **not** executed.

Is there are a good reason to follow a different strategy here?

Reply:
I think there are entirely different things. In the server world, an exception mapper maps an exception to a response.
For it, it may make sense to trigger response filters and react to the response.

In the client, we map a response to an exception that should be thrown to the user.

Or am I missing something?


Anyhow, the reason for the change is that if response exception filters are thrown on an exception when running response exception filters, they are triggered twice.

Reply:
I guess what you say makes sense, but I am not 100% sure yet.
But let's get it in and see in practice since it's obviously better than what we have